soroban-sdk/src/env.rs

```diff
@@ -532,19 +547,23 @@ impl Env {
         let auth_snapshot_in_hook = auth_snapshot.clone();
         env_impl
             .set_top_contract_invocation_hook(Some(Rc::new(move |host, event| {
-                if let ContractInvocationEvent::Finish = event {
-                    let new_auths = host
-                        .get_authenticated_authorizations()
-                        // If an error occurs getting the authenticated authorizations
-                        // it means that no auth has occurred.
-                        .unwrap_or_default();
-                    (*auth_snapshot_in_hook).borrow_mut().0.extend(new_auths);
+                match event {
+                    ContractInvocationEvent::Start => {}
+                    ContractInvocationEvent::Finish => {
+                        let new_auths = host
+                            .get_authenticated_authorizations()
+                            // If an error occurs getting the authenticated authorizations
```

Comment:
This comment is redundant now (and it's also incorrect; this simply shouldn't ever happen unless hook is bugged).

Reply:
I'll replace it with a simple unwrap.